server/datastore/mysql/policies.go

```diff
@@ -1663,3 +1663,32 @@ func (ds *Datastore) GetTeamHostsPolicyMemberships(
 
 	return hosts, nil
 }
+
+// GetPoliciesBySoftwareTitleID returns the policies that are associated with a
+// software title.
+func (ds *Datastore) GetPoliciesBySoftwareTitleID(
+	ctx context.Context,
+	softwareTitleID uint,
+	teamID *uint,
+) ([]fleet.AutomaticInstallPolicy, error) {
+	query := `
+	SELECT
+		p.id,
+		p.name
+	FROM policies p
+	JOIN software_installers si ON p.software_installer_id = si.id
+	JOIN software_titles st ON si.title_id = st.id
+	WHERE st.id = ?`
+
+	args := []any{softwareTitleID}
+	if teamID != nil {
+		query += " AND p.team_id = ?"
```

Comment:
This adds the `team_id` condition only if `teamID` is not `nil`, but I assume that the function argument `teamID == nil` mean look for a policy with no team? If so, the query should have the condition ` AND p.team_id IS NULL ` if `teamID == nil`, otherwise it could return the policy for any team.

Reply:
Oh, but this one really needs to be addressed, forgot about it when I wrote my top-level comment!

Reply:
@mna looks like we are not ever setting `policies.team_id` to `NULL` (if it's a policy for no team, it's set to `0`), so I think something like this would work yeah?


```go
	var tmID uint
	if teamID != nil {
		tmID = *teamID
	}
```

and then add `AND p.team_id = ?` to the query var above

Reply:
I'm not the most familiar with `policies`, but the table is defined with that column as nullable, and it seems at least in some cases we do insert without specifying the `team_id` (so, it stays `NULL`), e.g. https://github.com/fleetdm/fleet/blob/22840cfb2ca82e2e7cb799b8335723ac02e9ab43/server/datastore/mysql/policies.go#L56

However this is in a method called `NewGlobalPolicy`, so maybe we differentiate "No team" is `team_id = 0` and "Global" is `team_id = NULL`? If so I agree that your proposal would do the trick.